packages/11ty/_layouts/table-of-contents.11ty.js

```diff
@@ -35,63 +33,22 @@ module.exports = class TableOfContents {
         </div>
         `
       : ''
-    const containerClass = className === 'grid' ? 'is-fullhd' : ''
-    const contentsListClass = ['abstract', 'brief', 'grid'].includes(className)
-      ? className
+    const containerClass = type === 'grid' ? 'is-fullhd' : ''
+    const contentsListClass = ['abstract', 'brief', 'grid'].includes(type)
+      ? type
       : 'list'
 
-
-    let renderedSection
-
-    const defaultContentsItemParams = {
-      className,
-      config,
-      imageDir
-    }
-
-    const listItems = pages
-      .filter((page) => 
-        page.url !== tocPage.url
-        && (!section || section && page.data.section === section)
-        && page.data.type !== 'data' 
-        && page.data.toc !== false)
-      .map((page) => {
-        let listItem = ''
-        if (page.data.layout === 'table-of-contents' && page.data.section !== renderedSection) {
-          renderedSection = page.data.section 
-
-          const subPages =
-            config.params.tocType === 'full'
-              ? pages
-                  .filter(
-                    (item) =>
-                      item.data.section === page.data.section &&
-                      item.data.layout !== 'table-of-contents'
-                  )
-                  .map((item) => {
-                    if (page.fileSlug !== item.fileSlug)
-                      return `
-                        <li class="page-item">
-                          ${this.tableOfContentsItem({ ...defaultContentsItemParams, page: item })}
-                        </li>
-                      `
-                  })
-              : []
-
-          return `
-            <li class="section-item">
-              ${this.tableOfContentsItem({ ...defaultContentsItemParams, page })}
-              <ul>${subPages.join('')}</ul>
-            </li>
-          `
-        } else if (section || !page.data.section) {
-          return `
-            <li class="page-item">
-              ${this.tableOfContentsItem({ ...defaultContentsItemParams, page })}
-            </li>
-          `
-        }
-      })
+    /**
+     * The pages to include in the table of contents
+     * Either the project or a project section
+     */
+    const currentNavigationItem = this.eleventyNavigation(collections.tableOfContents).find(
+      ({ url }) => url === page.url
+    )
+    const isSection = !!currentNavigationItem.children && currentNavigationItem.children.length > 0
```

Comment:
Can this be written as below?
```suggestion
    const isSection = currentNavigationItem.children && currentNavigationItem.children.length
```

Reply:
not if we want it to evaluate to true or false, your suggestion will evaluate to the the length of the array

Reply:
For whatever reason checking for length greater than 0 seems to set a boolean has always struct me as odd, if you think either of below are less so, otherwise it may be my thinking that is odd
```
!!currentNavigationItem.children && !!currentNavigationItem.children.length
```
or
```
!!(currentNavigationItem.children && currentNavigationItem.children.length)
```